package/base-files/files/lib/functions/uci-defaults.sh

```diff
@@ -577,6 +577,14 @@ ucidef_set_ntpserver() {
 	json_select ..
 }
 
+ucidef_set_primarynetif() {
+	local netif="$1"
+
+	json_select_object system
+		json_add_string primarynetif "$netif"
+	json_select ..
+}
+
```

Comment:
This patch can be dropped entirely, as MAC address will come from get_mac_binary directly (which internally uses something similar ...)